### PR TITLE
Sqlite enable common extensions

### PR DESCRIPTION
**Pull request by joncfoo**
Before submitting your PR, check that you've:

- [x] Bumped the version number

After submitting your PR:

- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by joncfoo** [comment]
note: this builds on #789

**Comment by psibi** [comment]
Have you applied all the workaround patch as described here: https://github.com/yesodweb/persistent/blob/master/development.md#updating-sqlite-c-source-in-the-repo ?

**Comment by joncfoo** [comment]
cbits were not changed. Only build options that affect sqlite extensions were added.

**Comment by joncfoo** [comment]
Commiting this will require a rebase onto master after #789 is committed.

**Comment by joncfoo** [comment]
I'v rebased this branch.  Ready for review again.

File: persistent-sqlite/persistent-sqlite.cabal [hunk 1]
```diff
@@ -23,6 +23,18 @@ flag use-pkgconfig
...
+  default: True
+flag urifilenames
+  description: Enable URI filenames
+  default: True
+flag haveusleep
+  description: Enable use of usleep
+  default: True
+flag json1
+  description: Enable json1 extension
+  default: True
```
[the remark this answers is not in the document]
Comment (psibi) [reply persistent-sqlite/persistent-sqlite.cabal]: Sounds good to me! Thanks for the PR.

[the remark this answers is not in the document]
Comment (joncfoo) [reply persistent-sqlite/persistent-sqlite.cabal]: Happy to contribute :)

File: persistent-sqlite/persistent-sqlite.cabal [hunk 2]
```diff
@@ -23,6 +23,18 @@ flag use-pkgconfig
 flag build-sanity-exe
   description: Build a sanity check test executable
   default: False
+flag fulltextsearch
+  description: Enable full-text search
+  default: True
+flag urifilenames
+  description: Enable URI filenames
+  default: True
+flag haveusleep
```
Comment (MaxGabriel) [inline persistent-sqlite/persistent-sqlite.cabal]: Is there any convention for how flags are named in cabal files? I was wondering if there should be a preference for e.g. `have-usleep` instead. So far I'm not sure if there is any convention

Comment (joncfoo) [reply persistent-sqlite/persistent-sqlite.cabal]: I'm not sure if there's a convention either =\

How about adding a `cbits-` prefix so it's clear these are cbits related? e.g. `cbits-fulltextsearch`, `cbits-urifilenames`, `cbits-usleep`, `cbits-json1`

Comment (MaxGabriel) [reply persistent-sqlite/persistent-sqlite.cabal]: Nah it's fine. I don't think `cbits` means anything to people consuming persistent-sqlite anyway

**Comment by MaxGabriel** [comment]
@joncfoo Released!

I looked at some other Haskell packages and took the liberty of changing the flag names to be separated by `-`—there wasn't much of a consensus but that seemed like the most popular way to name flags.

**Comment by joncfoo** [comment]
Thanks @MaxGabriel!

**Pull request merged**